catt/cli.py

```diff
@@ -103,28 +107,111 @@ def write_config(settings):
         raise CattCliError("No device specified.")
 
 
+def hunt_subtitle(video):
+    """"Searches for subtitles in the current folder"""
+    video_path = Path(video)
+    video_path_stem_lower = video_path.stem.lower()
+    for entry in video_path.parent.iterdir():
+        entry_path = Path(entry)
+        if entry_path.is_dir():
+            continue
+        if entry_path.stem.lower().startswith(video_path_stem_lower) and \
+                entry_path.suffix.lower() in [".vtt", ".srt"]:
+            return str(entry_path.resolve())
+    return None
+
+
+def convert_srt_to_webvtt_helper(content):
+    content = re.sub(r'([\d]+)\,([\d]+)', r'\1.\2', content)
+
+    with tempfile.NamedTemporaryFile(mode='w+b',
+                                     suffix=".vtt",
+                                     delete=False) as vttfile:
+        target_filename = vttfile.name
+        vttfile.write("WEBVTT\n\n".encode())
+        vttfile.write(content.encode())
+        return target_filename
+
+
+def convert_srt_to_webvtt(filename):
+    # print("Converting {} to WebVTT".format(filename))
+    for possible_encoding in ['utf-8', 'iso-8859-15']:
+        try:
+            with open(filename, 'r', encoding=possible_encoding) as srtfile:
+                content = srtfile.read()
+                return convert_srt_to_webvtt_helper(content)
+        except UnicodeDecodeError:
+            pass
+    raise CattCliError("Could not find the proper encoding of {}. Please convert it to utf-8".format(filename))
+
+
+def load_subtitle_if_exists(subtitle, video, local_ip, port):
+    subtitle = subtitle or hunt_subtitle(video)
+    if subtitle is None:
+        return None
+    click.echo("Using subtitle {}".format(subtitle))
+
+    if "://" in subtitle:
+        # it's an URL
+        if subtitle.lower().endswith(".srt"):
+            content = requests.get(subtitle).text
+            subtitle = convert_srt_to_webvtt_helper(content)
+        else:
+            return subtitle
+
+    if subtitle.lower().endswith(".srt"):
+        subtitle = convert_srt_to_webvtt(subtitle)
```

Comment:
Shouldn't this give an error unless the extension is `.srt` or `.vtt` instead? We wouldn't want to continue with any arbitrary file.

Reply:
According to https://developers.google.com/cast/docs/media , Chromecast supports:

    TTML - Timed Text Markup Language
(example, https://gist.github.com/anotherhale/676a72edc84ca3a37c0c )
    WebVTT - Web Video Text Tracks
(example: https://github.com/brenopolanski/html5-video-webvtt-example/blob/master/MIB2-subtitles-pt-BR.vtt
    Line 21 Data Services
example: https://en.wikipedia.org/wiki/EIA-608

I confess I never heard about any of these 3 and I bet nobody will ever use cast with the latter. And XML is a bitch extension, so I think we are fine doing what the user tells us to instead of complaining.

What do you think ?

Reply:
The problem is that your code is not failing if the specified subtitle file has neither an ".srt" or ".vtt" extension.
(you are right that we should not concern ourselves with the other two obscure formats)

Reply:
My point is there may be somebody who would like to use one of the obscure formats and if we arbitrarily forbid it because the extensions we are restricting for restriction's sake.

In the end of the day there is no downside in sending the wrong file to chromecast as an extension, so I think we should not restrict it

Reply:
Also, we don't check the extension for the main files we cast, so we should not bother with the subtitles either.

Reply:
Hmm, okay, this makes sense to me, thanks.